test/complement.js

```diff
@@ -17,4 +17,13 @@ describe('complement', function() {
     eq(f(12, 2, 6), true);
   });
 
+  it('accepts fantasy-land functors', function() {
+    function Just(x) { this.value = x; }
+    Just.prototype.map = function(f) { return new Just(f(this.value)); };
+    function Nothing() {}
+    Nothing.prototype.map = function() { return this; };
+    eq(R.complement(new Just(true)).value, false);
+    eq(R.complement(new Nothing()) instanceof Nothing, true);
+  });
```

Comment:
`R.complement` never had an issue with short-circuiting since it is unary but I added a test for FL compatibility anyways.